include/circt/Dialect/Seq/SeqOps.td

```diff
@@ -36,14 +36,20 @@ def CompRegOp : SeqOp<"compreg",
   let arguments = (ins
     AnyType:$input,
     ClockType:$clk,
-    StrAttr:$name,
+    OptionalAttr<StrAttr>:$name,
     Optional<I1>:$reset,
     Optional<AnyType>:$resetValue,
     Optional<AnyType>:$powerOnValue,
     OptionalAttr<InnerSymAttr>:$inner_sym
   );
   let results = (outs AnyType:$data);
-  let hasCustomAssemblyFormat = 1;
+  let assemblyFormat = [{
+    (`sym` $inner_sym^)? custom<ImplicitSSAName>($name) $input `,` $clk
+    (`reset` $reset^ `,` $resetValue)?
+    (`powerOn` $powerOnValue^)? attr-dict `:` type($data)
+    custom<OptionalTypeMatch>(ref(type($data)), ref($resetValue), type($resetValue))
+    custom<OptionalTypeMatch>(ref(type($data)), ref($powerOnValue), type($powerOnValue))
```

Comment:
That is pretty clever! I hadn't considered using a custom parser to get some of the optional typing going.